agents/loco_mc_agent.py

```diff
@@ -286,6 +290,16 @@ def perceive(self, force=False):
             self.memory.add_triple(subj=chat_memid, pred_text="has_logical_form", obj=logical_form_memid)
             # New chat, mark as unprocessed.
             self.memory.tag(subj_memid=chat_memid, tag_text="unprocessed")
+            # Send data to the dashboard timeline
+            hook_data = {
+                "name" : "perceive",
+                "speaker" : speaker, 
+                "time" : self.last_chat_time,
+                "chat" : chat, 
+                "preprocessed" : preprocessed_chat, 
+                "logical_form" : chat_parse,
+            }
+            self.memory.dispatch_signal.send(self.perceive, data=hook_data)
```

Comment:
similarly, why is `self.memory` exposing a public `dispatch_signal` that you are freely using in loco_mc_agent.py ?